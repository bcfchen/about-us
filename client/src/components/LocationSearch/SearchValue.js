import React from "react";
import Typography from '@material-ui/core/Typography';

const SearchValue = ({ value, children }) => {
    return (
        <div className="Select-value async-select-value">
            <span className="Select-value-label">
                <Typography variant="headline" gutterBottom>{value.label}</Typography>
                {children}
            </span>
        </div>
    );
}

export default SearchValue;