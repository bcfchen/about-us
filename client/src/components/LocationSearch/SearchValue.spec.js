import React from "react";
import { shallow } from "enzyme";
import SearchValue from "./SearchValue";

describe("<SearchValue/>", () => {
    let component = null,
        deafultProps = {
            value: { label: "myLabel" },
            children: <div className="child-div">Child Div</div>
        };

    let render = (props = deafultProps) => {
        component = shallow(<SearchValue {...props} />);
    };
    beforeEach(() => { render(); });

    describe("on page load", () => {
        it("should render correctly", () => {
            expect(component).toMatchSnapshot();
        });

        it("should contain children components", () => {
            expect(component.find(".child-div").exists()).toEqual(true);
        });

        it("should contain display label correctly", () => {
            expect(component.find("WithStyles(Typography)").dive().dive().text()).toEqual("myLabel");
        });
    });
});