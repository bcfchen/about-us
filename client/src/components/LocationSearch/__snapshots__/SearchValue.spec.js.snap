// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchValue/> on page load should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchValue
    value={
        Object {
            "label": "myLabel",
          }
    }
>
    <div
        className="child-div"
    >
        Child Div
    </div>
</SearchValue>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="Select-value-label"
>
        <WithStyles(Typography)
                gutterBottom={true}
                variant="headline"
        >
                myLabel
        </WithStyles(Typography)>
        <div
                className="child-div"
        >
                Child Div
        </div>
</span>,
      "className": "Select-value async-select-value",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            gutterBottom={true}
            variant="headline"
>
            myLabel
</WithStyles(Typography)>,
          <div
            className="child-div"
>
            Child Div
</div>,
        ],
        "className": "Select-value-label",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "myLabel",
            "gutterBottom": true,
            "variant": "headline",
          },
          "ref": null,
          "rendered": "myLabel",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Child Div",
            "className": "child-div",
          },
          "ref": null,
          "rendered": "Child Div",
          "type": "div",
        },
      ],
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="Select-value-label"
>
          <WithStyles(Typography)
                    gutterBottom={true}
                    variant="headline"
          >
                    myLabel
          </WithStyles(Typography)>
          <div
                    className="child-div"
          >
                    Child Div
          </div>
</span>,
        "className": "Select-value async-select-value",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              gutterBottom={true}
              variant="headline"
>
              myLabel
</WithStyles(Typography)>,
            <div
              className="child-div"
>
              Child Div
</div>,
          ],
          "className": "Select-value-label",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "myLabel",
              "gutterBottom": true,
              "variant": "headline",
            },
            "ref": null,
            "rendered": "myLabel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Child Div",
              "className": "child-div",
            },
            "ref": null,
            "rendered": "Child Div",
            "type": "div",
          },
        ],
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
